src/main/scala/vectorpipe/ProcessOSM.scala

```diff
@@ -211,13 +228,34 @@ object ProcessOSM {
     * @param elements DataFrame containing node, way, and relation elements
     * @return DataFrame containing geometries.
     */
-  def constructGeometries(elements: DataFrame): DataFrame = {
-    import elements.sparkSession.implicits._
+  def constructGeometries(input: DataFrame): DataFrame = {
+    import input.sparkSession.implicits._
+    val st_pointToGeom = org.apache.spark.sql.functions.udf { pt: jts.Point => pt.asInstanceOf[jts.Geometry] }
+
+    // The following type conversion needed in case input comes from Change source
+    val elements =
+      Try(input.schema
+               .find(_.name == "members")
+               .get
+               .dataType
+               .asInstanceOf[ArrayType]
+               .elementType
+               .asInstanceOf[StructType]
+               .find(_.name == "type")
+               .get) match {
```

Comment:
How is there not a cleaner way to do this?

The only simplification I see is to replace `.find(_.name == "type").get` with `("type")`.